src/Microsoft.Health.Fhir.Core/Features/Conformance/Serialization/DefaultOptionHashSetJsonConverter.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance.Serialization
+{
+    internal class DefaultOptionHashSetJsonConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            EnsureArg.IsNotNull(writer, nameof(writer));
+            EnsureArg.IsNotNull(value, nameof(value));
+            EnsureArg.IsNotNull(serializer, nameof(serializer));
+
+            dynamic defaultOption = ((dynamic)value).DefaultOption;
+
+            if (value is IEnumerable enumerable)
+            {
+                var exists = false;
+                object first = null;
+
+                foreach (object item in enumerable)
+                {
+                    if (first == null)
+                    {
+                        first = item;
+                    }
+
+                    if (defaultOption == item)
+                    {
+                        exists = true;
+                        break;
+                    }
+                }
+
+                if (!exists)
```

Comment:
Maybe a bit comment about why this logic exist would be nice. When would the default option won't exist and why the first item would be okay in that case?

Reply:
Sounds like even though the item is hashset, but it will only serialize one object? is that correct understanding?

Reply:
Yes, even though our server can support multiple options, the capability statement only has one (in these cases). In a later task I want to use these sets to create a JSON Schema that defines all the valid options you can use to override

Reply:
Also, perhaps the first item is not ok, perhaps it should throw an exception if the option is misconfigured.